ext/updater.go

```diff
@@ -352,27 +353,34 @@ func (u *Updater) StartServer(opts WebhookOpts) error {
 	}
 
 	var tls bool
-	if opts.CertFile == "" && opts.KeyFile == "" {
+	switch {
+	case opts.CertFile == "" && opts.KeyFile == "":
 		tls = false
-	} else if opts.CertFile != "" && opts.KeyFile != "" {
+	case opts.CertFile != "" && opts.KeyFile != "":
 		tls = true
-	} else {
+	default:
 		return ErrMissingCertOrKeyFile
 	}
 
+	opts.setDefaults()
+
+	ln, err := net.Listen(opts.ListenNet, opts.ListenAddr)
+	if err != nil {
+		return fmt.Errorf("failed to listen on %s:%s: %w", opts.ListenNet, opts.ListenAddr, err)
+	}
+
 	u.webhookServer = &http.Server{
-		Addr:              opts.GetListenAddr(),
+		Addr:              ln.Addr().String(),
```

Comment:
Actually, as webhookServer is not exported we could skip this line completely.
srv.Addr is not used at all by the http.Server (except for srv.ListenAndServe).

Reply:
Good spot, I'd be up for removing it if you are! No point having dead code in there.